src/main/java/lotto/domain/Rank.java

```diff
@@ -1,25 +1,49 @@
 package lotto.domain;
 
+import java.util.Arrays;
+
 public enum Rank {
 
+	MISS(0, 0),
 	FIFTH(3, 5000),
 	FOURTH(4, 50000),
 	THIRD(5, 1500000),
+	SECOND(5, 30000000),
 	FIRST(6, 2000000000);
 
-	private final int matchedNumber;
-	private final int winningAmount;
+	private final int countOfMatch;
+	private final Money winningAmount;
 
-	Rank(int matchedNumber, int winningAmount) {
-		this.matchedNumber = matchedNumber;
-		this.winningAmount = winningAmount;
+	Rank(int countOfMatch, int winningAmount) {
+		this.countOfMatch = countOfMatch;
+		this.winningAmount = new Money(winningAmount);
 	}
 
-	public int getMatchedNumber() {
-		return matchedNumber;
+	public int getCountOfMatch() {
+		return countOfMatch;
 	}
 
 	public int getWinningAmount() {
-		return winningAmount;
+		return winningAmount.value();
+	}
+
+	public static Rank checkRank(long countOfMatch, boolean bonusNumberMatch) {
+		if (countOfMatch < Rank.FIFTH.getCountOfMatch()) {
+			return Rank.MISS;
+		}
+		if (countOfMatch == Rank.SECOND.getCountOfMatch()) {
+			return secondOrThirdRank(bonusNumberMatch);
+		}
+		return Arrays.stream(Rank.values())
+			.filter(rank -> rank.getCountOfMatch() == countOfMatch)
+			.findFirst()
+			.get();
+	}
```

Comment:
이렇게 개선해볼 수도 있을 것 같아요 :)
```suggestion
	if (countOfMatch == Rank.SECOND.getCountOfMatch()) {
			return secondOrThirdRank(bonusNumberMatch);
		}
		return Arrays.stream(Rank.values())
			.filter(rank -> rank.getCountOfMatch() == countOfMatch)
			.findFirst()
			.orElse(MISS);
	}
```